### ISSUE TITLE
Some sample programs can't show usage instructions

### ISSUE THREAD

**Issue by redtangent** [Mbed-TLS/mbedtls#1120]
### Description
- Type: Bug 
- Priority:  Minor

---------------------------------------------------------------
## Bug

**OS**  
linux|windows

**mbed TLS build:**  
Version: All versions 
Configuration: n/a
Compiler and options (if you used a pre-built binary, please indicate how you obtained it):  n/a
Additional environment information:  n/a

The following sample programs all appear to have a line `if( argc == 0 )` before displaying usage instructions. The problem with that is unless `main()` is called via some means other than the shell, `argc` will always be greater than `0` as the first argument in `argv` is always the name of the program being executed. The line should be `if( argc == 1 )` if it's meant to display when no arguments have been passed. That may be a mistake though, as some of the programs can execute with no provided options.

```
programs/util/pem2der.c
programs/x509/cert_req.c
programs/x509/cert_app.c
programs/x509/cert_write.c
programs/x509/req_app.c
programs/x509/crl_app.c
programs/pkey/dh_genprime.c
programs/pkey/key_app.c
programs/pkey/gen_key.c
programs/pkey/key_app_writer.c
programs/ssl/ssl_client2.c
programs/ssl/ssl_server2.c
programs/ssl/ssl_mail_client.c
```

**Comment by RonEld** [comment]
Hi @redtangent Thank you for reporting this issue!
We will look into this issue

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-1797

**Comment by tom-cosgrove-arm** [comment]
@aditya-deshpande-arm Fixing this should also fix the issue I mentioned today: #6700
